lib/cErrors.ml

```diff
@@ -88,30 +86,31 @@ let register_additional_error_info (f : Exninfo.info -> (Pp.t option Loc.located
     all the handlers of a list, and finally a [bottom] handler if all
     others have failed *)
 
-let rec print_gen ~anomaly ~extra_msg stk (e, info) =
+let rec print_gen ~anomaly ~extra_msg stk e =
   match stk with
   | [] ->
     print_anomaly anomaly e
   | h::stk' ->
-    try
-      let err_msg = h e in
+    match h e with
+    | Some err_msg ->
       Option.cata (fun msg -> msg ++ err_msg) err_msg extra_msg
-    with
-    | Unhandled -> print_gen ~anomaly ~extra_msg stk' (e,info)
-    | any -> print_gen ~anomaly ~extra_msg stk' (any,info)
+    | None ->
+      print_gen ~anomaly ~extra_msg stk' e
 
 let print_gen ~anomaly (e, info) =
   let extra_info =
     try CList.find_map (fun f -> Some (f info)) !additional_error_info_handler
     with Not_found -> None
   in
-  let extra_msg, info = match extra_info with
-    | None -> None, info
-    | Some (loc, msg) ->
-      let info = Option.cata (fun l -> Loc.add_loc info l) info loc in
-      msg, info
+  let extra_msg = match extra_info with
+    | None -> None
+    | Some (loc, msg) -> msg
   in
-  print_gen ~anomaly ~extra_msg !handle_stack (e,info)
+  try
+    print_gen ~anomaly ~extra_msg !handle_stack e
+  with exn ->
+    (* exception in error printer *)
+    print_anomaly anomaly exn
```

Comment:
Now that you made that explicit, maybe wrap this with a header saying "<error in printer>" or something. Not mandatory though.

Reply:
You mean `""` ?

Reply:
Oh, I wrote a tag that was munched by Markdown. It said "\<exception in printer\>".

Reply:
Ok will add a message while the review comes in.